src/encoder/ffmpeg.cpp

```diff
@@ -59,7 +59,30 @@ class Options {
 
 namespace shar::codecs::ffmpeg {
 
-static AVCodec* select_codec(Logger& logger, const ConfigPtr& config){
+static AVCodecContext* make_context(const ConfigPtr& config, AVCodec* codec, Size frame_size, std::size_t fps) {
+
+  AVCodecContext* context = avcodec_alloc_context3(codec);
+  assert(context);
+
+  std::fill_n(reinterpret_cast<char*>(context), sizeof(AVCodecContext), 0);
+  const std::size_t kbits = config->get<std::size_t>("bitrate", 5000);
```

Comment:
Looks like `config` is used only to get `bitrate`, so `config` parameter of this function could be replaced by `bitrate`.